pkg/api/norman/customization/cluster/formatter.go

```diff
@@ -61,16 +77,40 @@ func (f *Formatter) Formatter(request *types.APIContext, resource *types.RawReso
 	resource.Links["shell"] = shellLink
 	resource.AddAction(request, v32.ClusterActionGenerateKubeconfig)
 	resource.AddAction(request, v32.ClusterActionImportYaml)
+
 	if _, ok := resource.Values["rancherKubernetesEngineConfig"]; ok {
+
 		resource.AddAction(request, v32.ClusterActionExportYaml)
-		resource.AddAction(request, v32.ClusterActionRotateCertificates)
-		if rotateEncryptionKeyEnabled(f.clusterLister, resource.ID) {
-			resource.AddAction(request, v32.ClusterActionRotateEncryptionKey)
-		}
-		if _, ok := values.GetValue(resource.Values, "rancherKubernetesEngineConfig", "services", "etcd", "backupConfig"); ok {
+
+		// If a user has the BackupEtcd role/privilege, add it
+		if canUserBackupEtcd(request) {
 			resource.AddAction(request, v32.ClusterActionBackupEtcd)
-			resource.AddAction(request, v32.ClusterActionRestoreFromEtcdBackup)
 		}
+
+		// If user has permissions to update the cluster
+		if err := request.AccessControl.CanDo(v3.ClusterGroupVersionKind.Group, v3.ClusterResource.Name, "update", request, resource.Values, request.Schema); err == nil {
+			if _, ok := values.GetValue(resource.Values, "rancherKubernetesEngineConfig", "services", "etcd", "backupConfig"); ok {
+				resource.AddAction(request, v32.ClusterActionRestoreFromEtcdBackup)
+			}
+			resource.AddAction(request, v32.ClusterActionRotateCertificates)
+			if rotateEncryptionKeyEnabled(f.clusterLister, resource.ID) {
+				resource.AddAction(request, v32.ClusterActionRotateEncryptionKey)
+			}
+			if convert.ToBool(resource.Values["enableClusterMonitoring"]) {
+				resource.AddAction(request, v32.ClusterActionDisableMonitoring)
+				resource.AddAction(request, v32.ClusterActionEditMonitoring)
+			} else {
+				resource.AddAction(request, v32.ClusterActionEnableMonitoring)
+			}
```

Comment:
These fields can be enabled regardless of where the `rancherKubernetesEngineConfig` exists. However, in this change, these are moved inside an if-block so they will only be included for RKE1 clusters. These should be moved outside the if-block so they can be included regardless of whether `rancherKubernetesEngineConfig` field exists for the cluster.